icepyx/__init__.py

```diff
@@ -1,3 +1,17 @@
+from warnings import warn
+
+deprecation_msg = """icepyx v1.x is deprecated; the back-end systems on which it relies
+will be shut down as of late 2024. At that time, upgrade to icepyx v2.x, which uses the
+new NASA Harmony back-end, will be required. Please see
+<https://icepyx.readthedocs.io/en/latest/user_guide/changelog/v1.3.0.html> for more
+information!
+"""
+# IMPORTANT: This is being done before the other icepyx imports because the imported
+# code changes warning filters. If this is done after the imports, the warning won't
+# work.
+warn(deprecation_msg, DeprecationWarning, stacklevel=2)
```

Comment:
Sure about using [`DeprecationWarning`](https://docs.python.org/3/library/exceptions.html#DeprecationWarning) instead of [`PendingDeprecationWarning`](https://docs.python.org/3/library/exceptions.html#PendingDeprecationWarning)? There's also [`FutureWarning`](https://docs.python.org/3/library/exceptions.html#FutureWarning) that is more intended for end users :upside_down_face:

Reply:
Good question, not completely sure! RE: PendingDeprecationWarning:

>This class is rarely used as emitting a warning about a possible upcoming deprecation is unusual, and [DeprecationWarning](https://docs.python.org/3/library/exceptions.html#DeprecationWarning) is preferred for already active deprecations.

From PEP387:

>`PendingDeprecationWarning` may be used in special cases where the old and new versions of the API will coexist for many releases

I guess that provides a bit more clarity!

I'm still not sure about `FutureWarning`. Any examples of how other major projects handle similar scenarios?